subgraphs/truefi/src/entities/protocol.ts

```diff
@@ -0,0 +1,321 @@
+  financialsSnapshot.save();
+}
+
+export function addProtocolRepayVolume(
+  event: ethereum.Event,
+  amountUSD: BigDecimal
+): void {
+  const protocol = getOrCreateLendingProtocol();
+  const financialsSnapshot = getOrCreateFinancialsSnapshot(event, protocol);
+  financialsSnapshot.dailyRepayUSD =
+    financialsSnapshot.dailyRepayUSD.plus(amountUSD);
+  financialsSnapshot.save();
+}
+
+export function updateProtocolTVL(
+  event: ethereum.Event,
+  tvlChangeUSD: BigDecimal,
+  totalDepositBalanceUSDChange: BigDecimal
+): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.totalValueLockedUSD =
+    protocol.totalValueLockedUSD.plus(tvlChangeUSD);
+  protocol.totalDepositBalanceUSD = protocol.totalDepositBalanceUSD.plus(
+    totalDepositBalanceUSDChange
+  );
+  protocol.save();
+  const financialsSnapshot = getOrCreateFinancialsSnapshot(event, protocol);
+  financialsSnapshot.save();
+}
+
+export function updateProtocolBorrowBalance(
+  event: ethereum.Event,
+  bbChangeUSD: BigDecimal
+): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.totalBorrowBalanceUSD =
+    protocol.totalBorrowBalanceUSD.plus(bbChangeUSD);
+  protocol.save();
+  const financialsSnapshot = getOrCreateFinancialsSnapshot(event, protocol);
+  financialsSnapshot.save();
+}
+
+export function incrementProtocolUniqueUsers(): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.cumulativeUniqueUsers += 1;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueDepositors(): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.cumulativeUniqueDepositors += 1;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueBorrowers(): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.cumulativeUniqueBorrowers += 1;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueLiquidators(): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.cumulativeUniqueLiquidators += 1;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueLiquidatees(): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.cumulativeUniqueLiquidatees += 1;
+  protocol.save();
+}
+
+export function incrementProtocolPositionCount(): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.cumulativePositionCount += 1;
+  protocol.openPositionCount += 1;
+  protocol.save();
+}
+
+export function decrementProtocolOpenPositionCount(): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.openPositionCount -= 1;
+  protocol.save();
+}
+
+export function updateProtocolRewardToken(
+  timestamp: BigInt,
+  rewardToken: RewardToken,
+  emissionRate: BigInt
+): void {
+  const protocol = getOrCreateLendingProtocol();
+  protocol.rewardToken = rewardToken.id;
+  protocol.rewardTokenEmissionsAmount = emissionRate.times(
+    BIGINT_SECONDS_PER_DAY
+  );
+  protocol.rewardTokenEmissionsUSD = BIGDECIMAL_ZERO;
```

Comment:
save protocol after this line not sure why. But we have run into weird scenarios where the entity doesn't save properly in instances like this when passed without saving first.

Reply:
Thanks! Changed in d99eb453ec77afb674d3a370e1d0c781708ed260